bitcask/bitcask.go

```
@@ -331,6 +355,14 @@ func (bc *Bitcask) openActiveFile() error {
 		return fmt.Errorf("failed to open active file: %w", err)
 	}
 
+	// Seek to end so that Seek(0, io.SeekCurrent) in Put returns the correct
+	// write offset. O_APPEND ensures writes go to EOF, but the read cursor
+	// starts at 0 on macOS/POSIX.
+	if _, err := file.Seek(0, io.SeekEnd); err != nil {
+		file.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `file.Close` is not checked

(errcheck)